client/components/line-chart/index.js

```diff
@@ -204,43 +204,130 @@ class LineChart extends Component {
+		// This is needed so we can bind mouse events on the whole svg
+		// otherwise they will only be fired over svg shapes.
+		// SVG2 could solve it with `pointer-events: bounding-box;`
+		// but it's only supported in chrome at the moment.
+		svg.attr( 'pointer-events', 'all' );
 		svg
-			.selectAll( 'circle' )
-			.on( 'mouseenter', function( point, index ) {
-				self.handleMouseEnterPoint( this, index );
-			} )
-			.on( 'mouseout', function( point, index ) {
-				self.handleMouseOutPoint( this, index );
-			} );
+			.append( 'rect' )
+			.attr( 'x', 0 )
+			.attr( 'y', 0 )
+			.attr( 'width', '100%' )
+			.attr( 'height', '100%' )
+			.attr( 'fill', 'none' );
 	};
 
 	drawChart = ( svg, params ) => {
 		this.drawAxes( svg, params );
 		this.drawLines( svg, params );
 		this.drawPoints( svg, params );
+		this.drawOverlayElement( svg, params );
 		this.bindEvents( svg, params );
 		this.setState( { svg } );
 	};
 
-	handleMouseEnterPoint = point => {
-		d3Select( point ).attr( 'r', Math.floor( POINTS_SIZE * 1.5 ) );
+	handleMouseMove = ( X, Y, params ) => {
+		const { xScale, yScale } = params;
+		const { svg, data } = this.state;
+
+		const xDate = xScale.invert( X );
+		let closestDate = 0,
+			prevClosestDate = 0,
+			nextClosestDate = 0;
+
+		const firstDataSerie = data[ 0 ];
+		const drawFullSeries = firstDataSerie.length < POINTS_MAX;
+		// assume sorted by date
+		firstDataSerie.forEach( ( datum, index ) => {
+			if ( Math.abs( xDate - datum.date ) < Math.abs( xDate - closestDate ) ) {
+				closestDate = datum.date;
+				prevClosestDate = firstDataSerie[ index - 1 ]
+					? firstDataSerie[ index - 1 ].date
+					: datum.date;
+				nextClosestDate = firstDataSerie[ index + 1 ]
+					? firstDataSerie[ index + 1 ].date
+					: datum.date;
+			}
+		} );
 
-		this.setState( { pointHovered: point } );
-	};
+		const startDateBar = closestDate + Math.round( ( prevClosestDate - closestDate ) / 2 );
+		const endDateBar = closestDate + Math.round( ( nextClosestDate - closestDate ) / 2 );
 
-	handleMouseOutPoint = point => {
-		d3Select( point ).attr( 'r', POINTS_SIZE );
+		const bar = svg.select( `rect.line-chart__date-range-${ closestDate }` );
 
-		this.setState( { pointHovered: null } );
+		if ( bar.empty() ) {
+			svg
+				.select(
+					`rect.line-chart__date-range-selected:not(.line-chart__date-range-${ closestDate })`
+				)
+				.remove();
+
+			svg
+				.append( 'rect' )
+				.attr( 'class', `line-chart__date-range-selected line-chart__date-range-${ closestDate }` )
+				.attr( 'x', xScale( startDateBar ) )
+				.attr( 'y', 0 )
+				.attr( 'width', xScale( endDateBar ) - xScale( startDateBar ) )
+				.attr( 'height', yScale( 0 ) );
+
+			svg.selectAll( `circle.line-chart__line-point` ).attr( 'r', POINTS_SIZE );
+
+			if ( drawFullSeries ) {
+				const selectedPoints = svg.selectAll(
+					`circle.line-chart__line-point[cx="${ xScale( closestDate ) }"]`
+				);
+				selectedPoints.attr( 'r', Math.floor( POINTS_SIZE * 1.5 ) );
```

Comment:
Should `1.5` be it's own constant? maybe `POINT_HIGHLIGHT_SIZE_FACTOR` or something similar?

Reply:
👍